### PR TITLE
Added expansion query param support

### PR DESCRIPTION
**Pull request by Thought-Weaver**
## Summary

This adds the expansion into the query parameter for the route so it appears as part of the URL and allows the user's journey through the website to be a consistent experience; now when you navigate to a streamer's reviews, refresh the page, etc. it'll stay on the same expansion you have selected.

## Screenshots/Video

https://github.com/mateuscechetto/hearthstone-set-review-bot/assets/10130985/26d04bd8-a8ca-496a-a0fc-f8a3b2e18331

## Testing

Validate that the expansion is preserved in all navigation cases and that there aren't any side effects from the interaction between the service value and the query parameter.

#### Basics

- [x] Does `npm run build` succeed?
- [x] Does `npm run test` succeed? (More than before)

#### Testing

- [x] Did you create new tests and/or update existing tests?
- [x] Did you validate that your changes work hosted locally?

#### Accessibility (UX)

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +80,7 @@ The backend will be served at http://localhost:5000
 In another terminal run the frontend:
 
     cd client
-    ng serve
+    npm start
```
Comment (mateuscechetto) [inline README.md:83]: 👍

File: client/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const { pathsToModuleNameMapper } = require('ts-jest');
```
Comment (mateuscechetto) [inline client/jest.config.js:1]: Why was this change needed? (naive question, I'm still learning about setting up environments)

File: client/src/app/shared/data-access/expansion/expansion.service.ts [hunk 3]
```diff
@@ -15,17 +16,27 @@ export class ExpansionService {
...
+  setActiveExpansion(expansion: string) {
+    this.state.next(expansion);
+    this.updateQueryParam();
+  }
+
+  private updateQueryParam() {
+    this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams: { expansion: this.state.value },
+      queryParamsHandling: 'merge'
```
Comment (mateuscechetto) [inline client/src/app/shared/data-access/expansion/expansion.service.ts:39]: Clean!

**Review by mateuscechetto — APPROVED** [review]
Good job! I'll add the changes for the failing tests (as commented on discord) and merge it!

File: client/jest.config.js (hunk 2 above)
Comment (Thought-Weaver) [reply client/jest.config.js:1]: We need the mapper otherwise Jest gets confused about the tsconfig paths (it can't resolve things like "@shared/data-access/user/user.service" on its own.
